pkg/utils/powers_of_two.go

```diff
@@ -0,0 +1,70 @@
+package utils
+
+// RoundUpPowerOfTwo returns the next power of two greater than or equal to input.
+func RoundUpPowerOfTwo(input int) int {
+	k := 1
+	for k < input {
+		k = k << 1
+	}
+	return k
+}
+
+// RoundDownPowerOfTwo returns the next power of two less than or equal to input.
+func RoundDownPowerOfTwo(v int) int {
+	c := RoundUpPowerOfTwo(v)
+	if c == v {
+		return c
+	}
+	return c / 2
+}
+
+// RoundUpPowerOfTwoU rounds input up to the next power of two.
+// Examples:
+//
+//	RoundUpPowerOfTwoU(1) = 2
+//	RoundUpPowerOfTwoU(2) = 4
+//	RoundUpPowerOfTwoU(5) = 8
+func RoundUpPowerOfTwoU(input uint64) uint64 {
```

Comment:
[suggestion]
instead of using U, I think we could use generics.

Reply:
Tried it out in https://github.com/celestiaorg/celestia-app/pull/875/commits/c407ce28cc931200fd5e11630be4dcec60aedbe2 

Pro: results in less code
Con: less readable especially for developers unfamiliar with generics